packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/app/{{project_name}}.dart

```diff
@@ -68,19 +75,53 @@ class _MyMaterialApp extends StatefulWidget {
 class __MyMaterialAppState extends State<_MyMaterialApp> {
   Locale? _locale;
   {{#enable_change_language}}
-  late StreamSubscription<LanguageModel> _languageSubscription; {{/enable_change_language}}
+  late StreamSubscription<LanguageModel> _languageSubscription; {{/enable_change_language}}{{#enable_pin_code}}
+  late SessionConfig _sessionConfig;{{/enable_pin_code}}
 
   @override
   void initState() {
     {{^enable_change_language}}
-    _locale = const Locale('en'); {{/enable_change_language}}
+    _locale = const Locale('en'); {{/enable_change_language}}{{#enable_pin_code}}
+    _createSessionConfig();{{/enable_pin_code}}
     {{#enable_change_language}}
     _updateLocale(); {{/enable_change_language}} {{#push_notifications}}
     _configureFCM(); {{/push_notifications}}
     _configureInterceptors();
     super.initState();
   }
 
+  {{#enable_pin_code}}
+  void _createSessionConfig() {
+    context.read<UpdateAndVerifyPinBlocType>().events.deleteSavedData();
+    _sessionConfig = SessionConfig(
+      invalidateSessionForAppLostFocus: const Duration(seconds: 60),
+      invalidateSessionForUserInactivity: const Duration(seconds: 7),
+    );
+    context
+        .read<UpdateAndVerifyPinBlocType>()
+        .events
+        .setSessionState(SessionState.startListening);
+    _userInactivityListeners();
+  }
+
+  void _userInactivityListeners() {
+    _sessionConfig.stream.listen((timeoutEvent) {
+      if (timeoutEvent == SessionTimeoutState.userInactivityTimeout) {
+        context.read<RouterBlocType>().events.go(
+              const VerifyPinCodeRoute(),
+              extra: const PinCodeArguments(
+                  title: 'Enter Pin Code', isSessionTimeout: true),
+            );
+      } else if (timeoutEvent == SessionTimeoutState.appFocusTimeout) {
+        context.read<RouterBlocType>().events.go(
+              const VerifyPinCodeRoute(),
+              extra: const PinCodeArguments(
+                  title: 'Enter Pin Code', isSessionTimeout: true),
+            );
+      }
```

Comment:
This two bodies are exactly the same. Can their conditions be combined in one and use only one if statement?

Reply:
Yes, the second if was removed.